Tests/Realm.Tests/Database/AsyncTests.cs

```diff
@@ -426,5 +426,85 @@ public void SyncWrite_AsyncCommit_Mixed()
                 Assert.That(_realm.All<Person>().Single, Is.EqualTo(person));
             });
         }
+
+        [Test]
+        public void AsyncBeginWrite_CancelToken_OnAwait()
+        {
+            TestHelpers.RunAsyncTest(async () =>
+            {
+                var asyncThreadFactory = new AsyncContextThread().Factory;
+                using var cts = new CancellationTokenSource();
+                var tcs = new TaskCompletionSource<object>();
+                var taskCancelled = false;
+
+                var syncTask = Task.Run(() => AsyncContext.Run(() =>
+                {
+                    using var realm = GetRealm(_realm.Config);
+                    var transaction = realm.BeginWrite();
+
+                    tcs.TrySetResult(null);
+
+                    cts.Cancel();
+                    transaction.Rollback();
+                }));
```

Comment:
Can you try to see if this fixes the windows issue:

```suggestion
                var syncTask = Task.Run(() => AsyncContext.Run(async () =>
                {
                    using var realm = GetRealm(_realm.Config);
                    using var transaction = realm.BeginWrite();

                    tcs.TrySetResult(null);

                    cts.Cancel();
                    await Task.Delay(1000);
                }));
```